addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -0,0 +1,577 @@
+
+        /*
+        // Only IO devices report its state
+        if (!url.startsWith("io://")) {
+            return;
+        }
+
+        int state = getState(url);
+        if (state == -1) {
+            //relogin
+            login();
+            state = getState(url);
+        } else if (state == -2) {
+            //RTS device
+            return;
+        }
+
+        updateState(channelUID, new PercentType(state));
+        */
+
+        if(handler.getStateName() != null) {
+            State state = getState(handler, url);
+            if( state == null || state.equals(UnDefType.UNDEF)) {
+                return;
+            }
+
+            if( state.equals(UnDefType.NULL)) {
+                //relogin
+                login();
+                state = getState(handler, url);
+            }
+
+            updateState(channelUID, state);
+        }
+    }
+
+    private void logout() {
+        try {
+            sendToTahomaWithCookie(TAHOMA_URL + "logout");
+            cookie = "";
+        } catch (Exception e) {
+            logger.error("Cannot send logout command!");
+        }
+    }
+
+    public String readResponse(InputStream response) throws Exception {
+        String line;
+        StringBuilder body = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(response));
+
+        while ((line = reader.readLine()) != null) {
+            body.append(line).append("\n");
+        }
+        line = body.toString();
+        logger.trace("Response: {}", line);
+        return line;
+    }
+
+    public InputStream sendToTahomaWithCookie(String url) throws Exception {
+
+        URL cookieUrl = new URL(url);
+        HttpsURLConnection connection = (HttpsURLConnection) cookieUrl.openConnection();
+        connection.setDoOutput(false);
+        connection.setRequestMethod("GET");
+        setConnectionDefaults(connection);
+        connection.setRequestProperty("Cookie", cookie);
+
+        return connection.getInputStream();
+    }
+
+    public InputStream sendDataToTahomaWithCookie(String url, byte[] postData) throws Exception {
+
+        URL cookieUrl = new URL(url);
+        HttpsURLConnection connection = (HttpsURLConnection) cookieUrl.openConnection();
+        connection.setDoOutput(true);
+        connection.setRequestMethod("POST");
+        setConnectionDefaults(connection);
+        connection.setRequestProperty("Content-Length", Integer.toString(postData.length));
+        connection.setRequestProperty("Cookie", cookie);
+        try (DataOutputStream wr = new DataOutputStream(connection.getOutputStream())) {
+            wr.write(postData);
+        }
+
+        return connection.getInputStream();
+    }
+
+    public void setConnectionDefaults(HttpsURLConnection connection) {
+        connection.setInstanceFollowRedirects(false);
+        connection.setRequestProperty("User-Agent", TAHOMA_AGENT);
+        connection.setRequestProperty("Accept-Language", "de-de");
```

Comment:
don't know - I've checked web interface and en-US is used, so I changed it to en-US